client/lib/query-manager/index.js

```diff
@@ -104,6 +104,27 @@ export default class QueryManager {
 	}
 
 	/**
+	 * A sorting function that defines the sort order of items under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Array}  keys  Keys to be sorted
+	 * @param  {Array}  items Items by which to sort
+	 * @param  {Object} query Query object
+	 * @return {Array}        Sorted keys
+	 */
+	sort( keys, items, query ) {
+		return keys.sort( ( keyA, keyB ) => {
```

Comment:
It's a shame there's no non-mutative way to sort an array, not even in Lodash*! So it seems assuming the implementing developer is aware of this, they'd know they're mutating anyways. With that in mind... maybe it's enough to add a clarifying comment and drop the `return`?

_*_ At least not any that work in the same way as `Array#sort`